.circleci/config.yml

```diff
@@ -92,6 +92,44 @@ jobs:
             # retry introduced in https://github.com/algolia/react-instantsearch/pull/3475
             retry yarn test --maxWorkers=4
 
+  test_unit_react_v17:
+    <<: *defaults
+    steps:
+      - checkout
+      - run: *install_yarn_version
+      - restore_cache: *restore_yarn_cache
+      - run: *run_yarn_install
+      - save_cache: *save_yarn_cache
+      - run:
+          name: Install React v17 dependencies
+          command: |
+            yarn add --force -DW \
+              react@17.0.2 \
+              @types/react@17.0.47 \
+              react-dom@17.0.2 \
+              @types/react-dom@17.0.17 \
+              react-test-renderer@17.0.2 \
+              @types/react-test-renderer@17.0.2 \
+              @testing-library/react@12.1.4
+      - run:
+          name: Unit Tests
+          command: |
+            retry() {
+              MAX_RETRY=2
+              n=0
+              until [ $n -ge $MAX_RETRY ]
+              do
+                  "$@" && break
+                  n=$[$n+1]
+              done
+              if [ $n -ge $MAX_RETRY ]; then
+                exit 1
+              fi
+            }
```

Comment:
Maybe this can be made as an alias and instead could be referenced before this command as something like: 
```yml
- add_retry_util: *add_retry_util
```

This would allow us to only have one definition for both jobs that use it. It might be necessary to export the function though, I'm not sure if the job steps are run in isolation or not with Circle CI.

Reply:
They seem to be in insolation according to the [**CI output**](https://app.circleci.com/pipelines/github/algolia/react-instantsearch/3021/workflows/0d5d919a-6a45-4895-b081-3a38252e5268/jobs/56244/parallel-runs/0/steps/0-110).

That's the commit I tried: https://github.com/algolia/react-instantsearch/compare/chore/ci-alias-retry.

Reply:
alternatively we could isolate the whole `retry jest tests`, but that seems a bit much

Reply:
Thanks for trying! It's fine as is for me.